pkg/engine/overlayCondition.go

```diff
@@ -1,174 +1,190 @@
 package engine
 
 import (
+	"fmt"
 	"reflect"
+	"strconv"
 
 	"github.com/golang/glog"
 	"github.com/nirmata/kyverno/pkg/engine/anchor"
 )
 
-func meetConditions(resource, overlay interface{}) bool {
+func meetConditions(resource, overlay interface{}) (string, error) {
+	return checkConditions(resource, overlay, "/")
+}
+
+// resource and overlay should be the same type
+func checkConditions(resource, overlay interface{}, path string) (string, error) {
 	// overlay has no anchor, return true
 	if !hasNestedAnchors(overlay) {
```

Comment:
hasNestedAnchors calls getAnchorsFromMap, which checks for condition and existence anchor.
The overlayCondition is called only for mutate, where we only support condition and add an anchor.

Reply:
Remove existence anchor check in getAnchorsFromMap as it is only used in mutation.